File: Sharpex2D Java/framework/debug/logging/LogEntry.java
package sharpex2d.framework.debug.logging;

import java.util.Date;

public class LogEntry {

	/**
	 * Initializes a new LogEntry class.
	 * @param message The Message.
	 * @param level The Level.
	 * @param date The Date.
	 */
	public LogEntry(String message, LogLevel level, Date date)
	{
		_message = message;
		_level = level;
		_time = date;
	}
	
	private String _message;
	private LogLevel _level;
	private Date _time;
	
	/**
	 * Gets the Message.
	 * @return String.
	 */
	public String getMessage()
	{
		return _message;
	}
	/**
	 * Gets the LogLevel.
	 * @return LogLevel.
	 */
	public LogLevel getLevel()
	{
		return _level;
	}
	/**
	 * Gets the Date.
	 * @return Date.
	 */
	public Date getDate()
	{
		return _time;
	}
}

File: Sharpex2D/Framework/Debug/Logging/LogEntry.cs
using System;

namespace Sharpex2D.Framework.Debug.Logging
{
    public class LogEntry
    {
        /// <summary>
        /// Initializes a new LogEntry class.
        /// </summary>
        /// <param name="message">The Message.</param>
        /// <param name="level">The Level.</param>
        /// <param name="time">The Time.</param>
        public LogEntry(string message, LogLevel level, DateTime time)
        {
            Message = message;
            Level = level;
            Time = time;
        }
        /// <summary>
        /// Gets the Message.
        /// </summary>
        public string Message { get; private set; }
        /// <summary>
        /// Gets the LogLevel.
        /// </summary>
        public LogLevel Level { get; private set; }
        /// <summary>
        /// Gets or sets the Time.
        /// </summary>
        public DateTime Time { set; get; }
    }
} 
